### PR TITLE
Added Java/JVM and JavaScript development tool installation scripts

### PR DESCRIPTION
**Pull request by GJKrupa**
## Description
Installation scripts for the following developer tools:
- Eclipse for Java Developers (Neon version)
- JetBrains ToolBox
- SDKMAN library/tool manager
  - Groovy
  - Gradle
  - Maven
- NVM Node.JS version manager
  - Grunt build tool
  - Bower library manager

## Motivation and Context
Provides a quick-start installation for common Java/JVM/JavaScript development tools
required to start working on Home Office projects.

## How Has This Been Tested?
- Tested on standard developer laptop by removing all pre-installed tools (and
  associated .bashrc/.bash_profile entries) and running the scripts, then re-running
  the scripts to ensure that pre-installed tools are not re-installed unnecessarily.
  
- Checked that all tools can be executed and functioning icons exist for all GUI tools
  in the Ubuntu Unity launcher.

- Code reviewed by Mat Perry.  Request additional review by merger.

## Screenshots (if appropriate):
N/A

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.

### CODE REVIEW

File: install/eclipse.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+# Installer for JetBrains Toolbox
+# This will allow the installation, upgrade and management of other JetBrains tools
+# Running the tool itself will automatically add an icon to the launcher
+
```
Comment (sublimino) [inline install/eclipse.sh:7]: We tend towards `unofficial bash strict mode` (http://redsymbol.net/articles/unofficial-bash-strict-mode/ - although this should be documented somewhere, will investigate) Could this use `set -euo pipefail` please?

File: install/eclipse.sh [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+CHECKSUM="7cd7c5a020bb4a8644fdaa1b4e3b1b9daab34edd17677e28b2cdd7934f2a6432320fbba241e4fb91349ae056405721522e4bfe29970a3934d49a3e8385558691"
+ICONFILE=~/.local/share/applications/eclipse.desktop
+
+if [ ! -x "$OUTDIR/java-neon" ]; then
+
+    mkdir -p $OUTDIR
+    mkdir -p $TMPDIR
+
+    curl -L http://www.mirrorservice.org/sites/download.eclipse.org/eclipseMirror/technology/epp/downloads/release/neon/2/eclipse-java-neon-2-linux-gtk-x86_64.tar.gz -o $TMPDIR/eclipse.tar.gz
+    if [ "`sha512sum $TMPDIR/eclipse.tar.gz`" == "$CHECKSUM  $TMPDIR/eclipse.tar.gz" ]; then
```
Comment (sublimino) [inline install/eclipse.sh]: Some bits from https://google.github.io/styleguide/shell.xml - `$(subshell command)` is preferred to `s for readability

File: install/eclipse.sh [hunk 3]
```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+# Installer for JetBrains Toolbox
+# This will allow the installation, upgrade and management of other JetBrains tools
+# Running the tool itself will automatically add an icon to the launcher
+
+TMPDIR=~/tools/tmp
```
Comment (sublimino) [inline install/eclipse.sh]: README uses ```$HOME/tools``` - perhaps that could be used here for clarity?

File: install/eclipse.sh [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        echo "[Desktop Entry]" > $ICONFILE
+        echo "Type=Application" >> $ICONFILE
+        echo "Name=Eclipse Neon" >> $ICONFILE
+        echo "Exec=$OUTDIR/current/eclipse %u" >> $ICONFILE
+        echo "Icon=$OUTDIR/current/icon.xpm" >> $ICONFILE
+        echo "StartupNotify=false" >> $ICONFILE
+        echo "Categories=Development;IDE;" >> $ICONFILE
+        echo "Terminal=false" >> $ICONFILE
+    else
+        echo "Invalid checksum for tarball.  Aborting."
```
Comment (sublimino) [inline install/eclipse.sh:37]: Should this be followed by `exit 1`?

File: install/eclipse.sh [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    fi
+
+    rm -r $TMPDIR
+else
+    echo "Eclipse Neon is already installed"
+    echo "To remove it and re-install run rm -rf $OUTDIR/java-neon"
+fi
+
```
Comment (sublimino) [inline install/eclipse.sh:46]: `exit 0`?

File: install/jetbrains_toolbox.sh [hunk 6]
```diff
@@ -7,16 +7,27 @@
 VERSION=1.1.2143
 TMPDIR=~/tools/tmp
```
Comment (sublimino) [inline install/jetbrains_toolbox.sh]: As above - `${HOME}` is more readable

File: install/jetbrains_toolbox.sh [hunk 7]
```diff
@@ -7,16 +7,27 @@
...
 
-curl -L https://download.jetbrains.com/toolbox/jetbrains-toolbox-${VERSION}.tar.gz -o $TMPDIR/jetbrains-toolbox.tar.gz
-cd $OUTDIR
-tar xvzf $TMPDIR/jetbrains-toolbox.tar.gz
-rm -f $OUTDIR/latest
-ln -s $OUTDIR/*$VERSION* $OUTDIR/latest
+    curl -L https://download.jetbrains.com/toolbox/jetbrains-toolbox-${VERSION}.tar.gz -o $TMPDIR/jetbrains-toolbox.tar.gz
 
-rm -r $TMPDIR
+    if [ "`sha256sum $TMPDIR/jetbrains-toolbox.tar.gz`" == "$CHECKSUM  $TMPDIR/jetbrains-toolbox.tar.gz" ]; then
```
Comment (sublimino) [inline install/jetbrains_toolbox.sh]: `$()` instead of `?

File: install/jetbrains_toolbox.sh [hunk 8]
```diff
@@ -7,16 +7,27 @@
...
+        mkdir -p $OUTDIR
+        cd $OUTDIR
+        tar xvzf $TMPDIR/jetbrains-toolbox.tar.gz
+        rm -f $OUTDIR/latest
+        ln -s $OUTDIR/jetbrains-toolbox-$VERSION $OUTDIR/latest
+    else
+        echo "Invalid checksum for tarball.  Aborting."
```
Comment (sublimino) [inline install/jetbrains_toolbox.sh:25]: `exit 1`?

File: install/jetbrains_toolbox.sh [hunk 9]
```diff
@@ -7,16 +7,27 @@
...
+    fi
 
-$OUTDIR/latest/jetbrains-toolbox &
+    rm -r $TMPDIR
+
+    $OUTDIR/latest/jetbrains-toolbox &
+else
+    echo "JetBrains Toolbox $VERSION is already installed"
+    echo "To remove it and re-install run rm -rf $OUTDIR/jetbrains-toolbox-$VERSION"
+fi
```
Comment (sublimino) [inline install/jetbrains_toolbox.sh:35]: `exit 0` at end?

File: install/nvm.sh [hunk 10]
```diff
@@ -5,7 +5,7 @@ fi
 
 if [ "`nvm --version`" == '' ]; then
```
Comment (sublimino) [inline install/nvm.sh]: `$()`?

**Review by sublimino — COMMENTED** [review]
Please use `unofficial BASH strict mode` from http://redsymbol.net/articles/unofficial-bash-strict-mode/ (i.e. `set -euo pipefail`). 

Added some style/consistency questions.

File: install/eclipse.sh (hunk 1 above)
Comment (GJKrupa) [reply install/eclipse.sh:7]: Applied strict mode to all scripts but need to turn it off quite frequently in the sdkman and nvm scripts as they source third-party scripts that don't comply.

File: install/eclipse.sh (hunk 3 above)
Comment (GJKrupa) [reply install/eclipse.sh]: Replaced all ~ with $HOME

File: install/eclipse.sh (hunk 2 above)
Comment (GJKrupa) [reply install/eclipse.sh]: Replaced all backtick executions with $(...)

File: install/jetbrains_toolbox.sh (hunk 6 above)
Comment (GJKrupa) [reply install/jetbrains_toolbox.sh]: Replaced all ~ with $HOME

File: install/eclipse.sh (hunk 4 above)
Comment (GJKrupa) [reply install/eclipse.sh:37]: Added exit 1 at all aborts and exit 0 at script end

File: install/eclipse.sh (hunk 5 above)
Comment (GJKrupa) [reply install/eclipse.sh:46]: Added exit 1 at all aborts and exit 0 at script end

File: install/jetbrains_toolbox.sh (hunk 8 above)
Comment (GJKrupa) [reply install/jetbrains_toolbox.sh:25]: Added exit 1 at all aborts and exit 0 at script end

File: install/jetbrains_toolbox.sh (hunk 9 above)
Comment (GJKrupa) [reply install/jetbrains_toolbox.sh:35]: Added exit 1 at all aborts and exit 0 at script end

File: install/jetbrains_toolbox.sh (hunk 7 above)
Comment (GJKrupa) [reply install/jetbrains_toolbox.sh]: Replaced all backtick executions with $(...)

File: install/nvm.sh (hunk 10 above)
Comment (GJKrupa) [reply install/nvm.sh]: Replaced all backtick executions with $(...)

**Comment by GJKrupa** [comment]
Addressed all review comments as suggested

**Review by sublimino — APPROVED** [review]
It might be worth extracting the `sdkman` and `nvm` application versions into variables, but I missed them the first time round so will merge without them. Thanks for the changes!

**Pull request merged**